3DAmsterdam/Assets/Stadsmodel/Buildings/BuildingTileManager.cs

```diff
@@ -1,372 +1,364 @@
+                volgnummer = int.Parse(mesh.name.Split('_')[2]);
             }
-            GameObject container = new GameObject(Meshnaam);
-            container.transform.parent = transform;
-            container.layer = LayerMask.NameToLayer("Panden");
-
-            //positioning container
-            Vector3RD hoekpunt = new Vector3RD(X, Y, 0);
-            double OriginOffset = 500;
-            if (btd.id.z>0)
-            {
-                OriginOffset = 500;
-            }
-            Vector3RD origin = new Vector3RD(hoekpunt.x + (OriginOffset), hoekpunt.y + (OriginOffset), 0);
-            Vector3 UnityOrigin = CoordConvert.RDtoUnity(origin);
-            container.transform.localPosition = UnityOrigin;
-            double Rotatie = CoordConvert.RDRotation(origin);
-            container.transform.Rotate(Vector3.up, (float)Rotatie);
-
-            //add mesh
-            MeshFilter mf = container.AddComponent<MeshFilter>();
-            mf.sharedMesh = ass;
-            MeshCollider mc = container.AddComponent<MeshCollider>();
-            mc.sharedMesh = ass;
-            //add material
-            MeshRenderer mr = container.AddComponent<MeshRenderer>();
-            mr.material = GebouwMateriaal;
-            yield return null;
-            //gebouwenlijst toevoegen
-            if (ObjectenlijstAanwezig)
-            {
-                gebouwenlijst = new Dictionary<float, string>();
-                
-                string[] panden = Regex.Split(bagidRegels[volgnummer * 2], ",");
-                for (int i = 0; i < panden.Length - 1; i++)
-                {
-
-                        gebouwenlijst.Add((float)i,panden[i]);
 
-                }
-                ObjectMapping Bid = container.AddComponent<ObjectMapping>();
-                Bid.Objectenlijst = gebouwenlijst;
-                Bid.DefaultMaterial = GebouwMateriaal;
-                Bid.HighlightMaterial = HighLightMateriaal;
-                Bid.SetMesh(ass);
+            GameObject container = CreateGameObjectWithMesh(mesh);
 
-            }
-            //add to Buildingtiledata
-            btd.Gameobjecten.Add(container);
-            
+            AddObjectIDs(mesh, volgnummer, objectIDs, container);
             
+            tileData.gameObjects.Add(container);
 
             yield return null;
         }
-        ActiveBuilds.Remove(btd.id);
-        //////instantiaten en assets loaden
-        btd.Status = BuildingTileStatus.Built;
+        ActiveBuilds.Remove(tileData.tileID);
+        tileData.Status = TileStatus.Built;
+
+
+    }
+
+    private GameObject CreateGameObjectWithMesh(Mesh mesh)
+    {
+        string gameObjectName = mesh.name;
+        float X = float.Parse(gameObjectName.Split('_')[0]);
+        float Y = float.Parse(gameObjectName.Split('_')[1]);
+        
+        GameObject container = new GameObject(gameObjectName);
+        container.transform.parent = transform;
+        container.layer = LayerMask.NameToLayer("Panden");
```

Comment:
Een LayerMask kan ook geserialized worden in Unity.
Wanneer je bijvoorbeeld een declareer bovenaan:
`[SerializeField]
private LayerMask layerMask;`

Dan kan je die vervolgens hier gebruiken dmv. `container.layer = layerMask;`

Het grote voordeel is dan dan de "Panden" naam niet meer hardcoded is. 
Zodra we dan de lagen zouden hernoemen blijft dit script gewoon werken.